frontend/src/TestEdit.js

```diff
@@ -0,0 +1,263 @@
+        const value = target.value;
+        const name = target.name;
+        let newItem = {...item};
+
+        newItem.test[name] = value;
+        setItem(newItem);
+    }
+
+
+    function handleCheckboxChange(id, sId, event) {
+        let newItem = {...item};
+        const target = event.target;
+
+        let q =newItem.questions.find(q => q.id === +id);
+        if(!target.checked){
+            (q.answers) = (q.answers).filter(item => item !== +sId );
+        }else {
+            if(!q.answers.includes(+sId)){
+                (q.answers).push(+sId);
+            }
+        }
+            setItem(newItem);
+        }
+
+    function handleQuestionsChange( id,  sId =null,event ) {
+        const value =  event.target.value;
+        let newItem = {...item};
+
+        if(sId===null){
+            newItem.questions[id].question = value;
+        }else {
+            (newItem.questions[id]).answerVariants[sId] = value;
+        }
+        setItem(newItem);
+    }
+
+
+    async function handleSubmit(event) {
+        event.preventDefault();
+        item.added= Array.from(item.added);
+        item.deleted= Array.from(item.deleted);
+
+        await fetch('/subload/tests' + (item.test.id ? '/' + item.test.id : ''), {
+            method: (item.test.id) ? 'PUT' : 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(item),
+        });
+    }
+
+
+    async function delAnswer(index, sIndex) {
+        let newItem = {...item};
+        let q =newItem.questions[index]
+
+        q.answerVariants=  (q.answerVariants).filter(i => i !== q.answerVariants[sIndex]);
+        setItem(newItem);
+        q.answers = (q.answers).filter(a => a !== +sIndex);
+        q.answers.forEach(function(a,index, thisArr) {
+            if(a > +sIndex){
+                thisArr[index]-=1;
+            }
+        });
+
+    }
+
+
+    async function addAnswer(index) {
+        let newItem = {...item};
+
+        (newItem.questions[index]).answerVariants.push([]);
+        setItem(newItem);
+    }
+
+
+    async function addQuestion() {
+        let newItem = {...item};
+        const que= {name:"", id:+(++(maxId)), answerVariants:[""],answers:[]};
+
+        newItem.questions.push(que);
+        newItem.added.add(que.id)
+        setItem(newItem);
+    }
+
+
+    function isChecked(id, sIndex){
+        return (item.questions.find(q => q.id === +id)).answers.includes(sIndex);
+    }
+
+
+    async function delQuestion(index) {
+        let newItem = {...item};
+        let questionId=newItem.questions[index].id
+
+        if(newItem.added.has(questionId)){
+            newItem.added.delete(questionId);
+        }else{
+            newItem.deleted.add(questionId);
+        }
+        newItem.questions=  newItem.questions.filter(i => i !== newItem.questions[index]);
+        setItem(newItem);
+    }
+
+
+    useEffect(() => {
```

Comment:
Consider rewriting this fetch using [react-query](https://tanstack.com/query/v3/docs/react/overview)

Reply:
I tried it, but in my case(fetch calls conditionally and `useState` is used ) it's not convenient. can we discuss this during the meeting?